include/zephyr/bluetooth/services/hrs.h

```diff
@@ -23,6 +23,25 @@
 extern "C" {
 #endif
 
+/** @brief Heart rate service callback structure */
+struct bt_hrs_cb {
+	/** @brief Heart rate notifications changed
+	 *
+	 * @param enabled Flag that is true if notifications were enabled, false
+	 *                if they were disabled.
+	 */
+	void (*ntf_changed)(bool enabled);
+};
+
+/** @brief Heart rate service callback register
+ *
+ * This function will register callbacks that will be called in
+ * certain events related to Heart rate service.
+ *
```

Comment:
This API definition does not cover the behavior of the implementation: That it will overwrite a previously registered listener.

I have a suggestion on how to tweak the definition to allow for it. See below.

```suggestion
 *
 * Limitations in the current version:
 *  - This function may be called only once per boot.
 *
```

Reply:
In such case it is better to implement it as a list of callbacks, similar to https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/scan.c#L1809-L1812.